cellprofiler/modules/exporttodatabase.py

```diff
@@ -1835,13 +2004,22 @@ def prepare_run(self, workspace, as_data_tool=False):
                 column_defs = self.get_pipeline_measurement_columns(pipeline,
                                                                     image_set_list)
                 if self.objects_choice != O_ALL:
-                    onames = [cpmeas.EXPERIMENT, cpmeas.IMAGE, cpmeas.NEIGHBORS]
+                    onames = [cellprofiler.measurement.EXPERIMENT,
+                              cellprofiler.measurement.IMAGE,
+                              cellprofiler.measurement.NEIGHBORS]
                     if self.objects_choice == O_SELECT:
                         onames += self.objects_list.selections
                     column_defs = [column for column in column_defs
                                    if column[0] in onames]
                 self.create_database_tables(self.cursor, workspace)
             return True
+        except sqlite3.OperationalError as err:
+            if err.message.startswith("too many columns"):
+                # Maximum columns reached
+                # https://github.com/CellProfiler/CellProfiler/issues/3373
+                message = "MySQL Error: maximum columns reached. \n" \
+                          "Try exporting a single object per table"
+                cellprofiler.gui.dialog.Error("Error", message)
```

Comment:
While we're at it, let's create a friendly message for other `OperationalError`s. E.g.,

```python
if err.message.startswith("too many columns"):
    # ...
else:
    cellprofiler.gui.dialog.Error("Error", "MySQL Error: " + err.message)
```